src/compact_strings.rs

```diff
@@ -489,6 +497,7 @@ impl Index<usize> for CompactStrings {
 /// assert_eq!(iter.next(), Some("Three"));
 /// assert_eq!(iter.next(), None);
 /// ```
+#[must_use = "Iterators are lazy and must be consumed to be useful"]
```

Comment:
Please change it to "iterators are lazy and do nothing unless consumed" to be consistent with std.